tests/trace/data_serialization/test_cases/core_type_factory.py

```diff
@@ -0,0 +1,23 @@
+import weave
+from weave.trace.objectify import register_object
+
+
+def make_op():
+    @weave.op
+    def say_hello(name: str):
+        return "hello " + name
+
+    return say_hello
+
+
+@register_object
+class CustomObject(weave.Object):
+    my_name: str
+
+    @weave.op
+    def say_hello(self):
+        return "hello " + self.my_name
+
+    @classmethod
+    def from_obj(cls, obj: weave.Object):
```

Comment:
The `obj` here is actually a `WeaveObject`, not `weave.Object` (yes, the nomenclature could use work...)

Reply:
👍